### PR TITLE
[port v2int 4.0]  Fix signal error event in case we reconnect and we miss the first signal

### PR DESCRIPTION
**Pull request by NicholasCouri**
…nal (#14939)

## Description

Fix regression caused by [Download snapshot with handle before throwing "Fetched snapshot is older than the received ack" by NicholasCouri · Pull Request #14225 · microsoft/FluidFramework ·
GitHub](https://github.com/microsoft/FluidFramework/pull/14225) in which, if we reconnect, we will send the SendError Logic. WIll work on adding a new test to prevent this error from happening again.

[How contribute to this repo](https://github.com/microsoft/FluidFramework/blob/main/CONTRIBUTING.md).

[Guidelines for Pull Requests](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

The sections included below are suggestions for what you may want to include.
Feel free to remove or alter parts of this template that do not offer value for your specific change.

## Description

> A concise description of the changes (bug or feature) and their impact/motivation.
> If this description is short enough to be used as the title, delete this section and just use the title.

> For bug fixes, also include specifics of how to reproduce it / confirm it is fixed.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

## Breaking Changes

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.
> See [Breaking-vs-Non-breaking-Changes](https://github.com/microsoft/FluidFramework/wiki/Breaking-vs-Non-breaking-Changes) for details.
> If there are no breaking changes, delete this section.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

> List any specific things you want to get reviewer opinions on, and anything a reviewer would need to know to review this PR effectively.
> Things you might want to include:
>
> -   Questions about how to properly make automated tests for your changes.
> -   Questions about design choices you made.
> -   Descriptions of how to manually test the changes (and how much of that you have done).
> -   etc.
>
> If you have any questions in this section, consider making the PR a draft until all questions have been resolved.

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1982,12 +1982,13 @@ export class ContainerRuntime
 					clientSignalSequenceNumber: envelope.clientSignalSequenceNumber,
 				});
 			} else if (
-				this.consecutiveReconnects === 0 &&
 				envelope.clientSignalSequenceNumber ===
-					this._perfSignalData.trackingSignalSequenceNumber
+				this._perfSignalData.trackingSignalSequenceNumber
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1986]: I am curious, why was this change added by https://github.com/microsoft/FluidFramework/pull/14225?

**Comment by agarwal-navin** [comment]
> …nal (#14922)

Can you please move this to the title which is incomplete :-).

**Comment by agarwal-navin** [comment]
Also, maybe remove the template part of the description.

**Comment by agarwal-navin** [comment]
Add [port v2int 4.0] to the title to make it clear its a port. Similar to https://github.com/microsoft/FluidFramework/pull/14915

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1986]: It was actually by mistake - it was not even related to that bug - I must have had that file checked out and missed it

**Review by markfields — APPROVED** [review]
Btw I don't think you should use branch name starting with `release/` for your own development.

**Comment by anthony-murphy** [comment]
is it possible to add a test for this? ideally all patches also have a regression test.

**Comment by NicholasCouri** [comment]
> is it possible to add a test for this? ideally all patches also have a regression test.

I talked to Tony and will proceed with the check in, and add the test ASAP
